staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured_test.go

```diff
@@ -58,3 +64,77 @@ func TestUnstructuredList(t *testing.T) {
 		t.Fatalf("unexpected fields: %#v", items[0])
 	}
 }
+
+func TestConversionRoundtrip(t *testing.T) {
+	testCases := []struct {
+		obj runtime.Object
+	}{
+		{
+			obj: &Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": "v1",
+					"kind":       "Foo",
+					"metadata": map[string]interface{}{
+						"name": "foo1",
+					},
+				},
+			},
+		},
+	}
+
+	for i := range testCases {
+		doRoundTrip(t, testCases[i].obj)
+		if t.Failed() {
+			break
+		}
+	}
+}
+
+// copy from staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/converter_test.go
```

Comment:
can we move and unify both into staging/src/k8s.io/apimachinery/pkg/conversion/unstructured/testing/roundtrip.go?

Reply:
Done. I moved the whole test file into a subdirectory - tests share type definitions (`A`, `B`, etc) so I couldn't move just the roundtrip test.